core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -172,6 +172,7 @@ protected boolean ignoreReplicaException(Throwable e) {
         if (isConflictException(e)) {
             return true;
         }
+        // TODO should we check here for refresh and flush failures which should not make replica fail? if so, which are they?
```

Comment:
I think it's OK to mark the shard as deleted - if there was some connectivity failure (i.e., the shard it self is not broken) *strictly* speaking the shard is no longer a good copy as it doesn't maintain the "all writes so far are visible to searches" semantics. I can see that it's potentially a big deal for nothing (after 1s it will be OK) but I think we should for now do the right thing and fail the shard?